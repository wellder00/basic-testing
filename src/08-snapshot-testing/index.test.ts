// Uncomment the code below and write your tests
import { generateLinkedList } from './index';

interface ListNode {
  value: string | null;
  next: ListNode | null;
}

describe('generateLinkedList', () => {
  let list1: string[];
  let result: ListNode;
  
  beforeEach(() => {

    list1 = [
      'apple',
      'banana',
      'orange',
      'grape',
      'melon',
      'pineapple',
      'pear',
      'strawberry',
      'kiwi',
      'blueberry',
    ];

    result = {
      next: {
        next: {
          next: {
            next: {
              next: {
                next: {
                  next: {
                    next: {
                      next: {
                        next: {
                          next: null,
                          value: null,
                        },
                        value: 'blueberry',
                      },
                      value: 'kiwi',
                    },
                    value: 'strawberry',
                  },
                  value: 'pear',
                },
                value: 'pineapple',
              },
              value: 'melon',
            },
            value: 'grape',
          },
          value: 'orange',
        },
        value: 'banana',
      },
      value: 'apple',
    };
  });
  test('should generate linked list from values 1', () => {
    const resultLinkedList = generateLinkedList(list1);
    expect(resultLinkedList).toStrictEqual(result);
  });

  test('should generate linked list from values 2', () => {
    const resultLinkedList = generateLinkedList(list1);
    expect(resultLinkedList).toMatchSnapshot();
  });
});
