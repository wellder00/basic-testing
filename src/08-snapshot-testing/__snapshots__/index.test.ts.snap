// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateLinkedList should generate linked list from values 2 1`] = `
{
  "next": {
    "next": {
      "next": {
        "next": {
          "next": {
            "next": {
              "next": {
                "next": {
                  "next": {
                    "next": {
                      "next": null,
                      "value": null,
                    },
                    "value": "blueberry",
                  },
                  "value": "kiwi",
                },
                "value": "strawberry",
              },
              "value": "pear",
            },
            "value": "pineapple",
          },
          "value": "melon",
        },
        "value": "grape",
      },
      "value": "orange",
    },
    "value": "banana",
  },
  "value": "apple",
}
`;
